### PR TITLE
Update dependencies and use worker v16

### PR DESCRIPTION
**Pull request by Ereski**
- Remove references to `jellyfish-test-harness`, `jellyfish-plugin-base`, `jellyfish-plugin-default`, and `jellyfish-plugin-product-os`.
- Limit usage of `jellyfish-types` to contract types.
- Remove all actions that were imported into the worker.
- Adapt to the new plugin interface.
- Remove unused dependencies.

Change-type: major
Signed-off-by: Carol Schulze <carol@balena.io>

### CODE REVIEW

**Comment by joshbwlng** [comment]
Noticed we have a number of actions that depend on type contracts provided by `plugin-default`, such as `action-ping` and the `ping` type. Maybe we should just move those into action library so they live side by side. Doing this should also allow us to remove the `plugin-default` dependency all together?

We could also update `createSupportThread()` calls to be more generic and instead call something like `createThread()`. This would require us to move these somewhat core type contracts into this repo and out of `plugin-default` though.

**Comment by Ereski** [comment]
@joshbwlng from what I can see so far the only action that needs the `support-thread` type is `action-broadcast`. For everything else I can just use `createContract` and `createEvent`.

**Comment by Ereski** [comment]
Would you prefer merging plugin-default with the action library now, or just move stuff around to make JF work sooner?

**Comment by joshbwlng** [comment]
@Ereski I'm fine with merging things later on. My main goal at the moment is to get things unblocked. My `plugin-default` PR (https://github.com/product-os/jellyfish-plugin-default/pull/908) seems to have a dependency on `action-library`. Also there seems to be some dependencies on `plugin-default` in `action-library` (`action-increment-tag` requires the `tag` type for example). So I'm not sure how to properly satisfy everything without merging them. I can try bringing the changes you've made here into my `plugin-default` PR unless you have something else in mind.

File: test/integration/actions/action-set-password.spec.ts [hunk 1]
```diff
@@ -52,19 +54,19 @@ describe('action-set-password', () => {
 
 	test('should change the password of a password-less user given no password', async () => {
 		const user = await ctx.createUser(
-			ctx.generateRandomWords(1),
+			coreTestUtils.generateRandomSlug(),
 			PASSWORDLESS_USER_HASH,
 		);
 
 		// TODO: temporary workaround for context/logContext mismatch
```
Comment (joshbwlng) [inline test/integration/actions/action-set-password.spec.ts:61]: Out of curiosity, do we have a plan for resolving this workaround?

**Comment by joshbwlng** [comment]
Merging this work into https://github.com/product-os/jellyfish-plugin-default/pull/908

**Pull request closed without merging**